cincinnati/src/plugins/internal/arch_filter.rs

```diff
@@ -51,6 +51,46 @@ impl ArchFilterPlugin {
 
         Ok(Box::new(plugin))
     }
+
+    pub fn remove_conditional_edges(
+        &self,
+        ce: &mut Vec<cincinnati::ConditionalEdge>,
+        arch: String,
+    ) -> Fallible<()> {
+        ce.into_iter()
+            .for_each(|ce: &mut cincinnati::ConditionalEdge| {
+                ce.edges
+                    .retain(|e| e.from.contains(&arch) || e.to.contains(&arch));
+            });
+
+        Ok(())
+    }
+
+    pub fn rm_ce_arch_identifier(
+        &self,
+        vce: &mut Vec<cincinnati::ConditionalEdge>,
+        arch: String,
+    ) -> Fallible<()> {
+        vce.into_iter()
+            .for_each(|ce: &mut cincinnati::ConditionalEdge| {
+                ce.mut_edges().into_iter().for_each(|e| {
+                    e.from = self.remove_arch_info(&e.from, arch.clone()).unwrap();
+                    e.to = self.remove_arch_info(&e.to, arch.clone()).unwrap();
+                });
+            });
+
+        Ok(())
+    }
+
+    fn remove_arch_info(&self, v: &String, arch: String) -> Result<String, Error> {
+        semver::Version::parse(v)
+            .context(v.clone())
+            .map(|mut version| {
+                version.build.retain(|elem| elem.to_string() != arch);
+                trace!("CE: rewriting version {} ->  {}", v, version);
```

Comment:
```suggestion
                trace!("CE: rewriting version {} -> {}", v, version);
```
world's smallest nitpick - removed extra space